cmd/iam-etcd-store.go

```diff
@@ -313,11 +318,31 @@ func (ies *IAMEtcdStore) loadUser(user string, isSTS bool, m map[string]auth.Cre
 	if u.Credentials.IsExpired() {
 		// Delete expired identity.
 		ctx := ies.getContext()
-		deleteKeyEtcd(ctx, ies.client, getUserIdentityPath(user, isSTS))
-		deleteKeyEtcd(ctx, ies.client, getMappedPolicyPath(user, isSTS, false))
+		deleteKeyEtcd(ctx, ies.client, getUserIdentityPath(user, userType))
+		deleteKeyEtcd(ctx, ies.client, getMappedPolicyPath(user, userType, false))
 		return nil
 	}
 
+	// If this is a service account, rotate the session key if we are changing the server creds
+	if globalOldCred.IsValid() && u.Credentials.IsServiceAccount() {
```

Comment:
globalOldCred - since the oldCred is always set there is potential for abuse if some caller refactors tries to use `loadUser`  - it is better for globalOldCred to become invalid as soon as the server starts.

Reply:
> globalOldCred - since the oldCred is always set there is potential for abuse if some caller refactors tries to use `loadUser` - it is better for globalOldCred to become invalid as soon as the server starts.

Done